pkg/volume/util/fsquota/quota_linux.go

```diff
@@ -303,6 +303,7 @@ func SupportsQuotas(m mount.Interface, path string) (bool, error) {
 // AssignQuota chooses the quota ID based on the pod UID and path.
 // If the pod UID is identical to another one known, it may (but presently
 // doesn't) choose the same quota ID as other volumes in the pod.
+//lint:ignore SA4009 poduid is overwritten by design, see comment below
```

Comment:
I think it's fine to use a local variable for poduid to fix the linter error

Reply:
From the comment on lines 317-321 it seems like the code is meant to be disabled but can be re-enabled by just removing line 322. Making `poduid` a local variable would break that behaviour.